### ISSUE TITLE
horizontal scroll bar doesn't response mouse side scroller

### ISSUE THREAD

**Issue by BeatMercy** [Microsoft/vscode#54024]
I'm a VS Code user with logic **Mx Master**, 
I used to using the side scroller to make horizontal scroll. 
### In Excel, it can response my side scroller. Columns rolls on every move of side scroller, but VS Code doesn't.
I hope this feature can be added by next release, thx!

**Comment by myfrom** [comment]
Can confirm.

Using MX Master 2S, horizontal scroll wheel on the side of the mouse works fine with other apps, yet in VS Code it works only after you minimize and bring back the window and stops working again when you switch to another window. Shift + scroll and touchpad scroll work fine all the time.

Windows 10 Home version 1803,
Code Version: 1.25.1
Commit: 1dfc5e557209371715f655691b1235b6b26a06be
Date: 2018-07-11T15:43:53.668Z
Electron: 1.7.12
Chrome: 58.0.3029.110
Node.js: 7.9.0
V8: 5.8.283.38
Architecture: x64

I've seen other issues about this closed as duplicates of an issue about smooth scrolling, this is not about smooth scrolling, please don't mark it as duplicate of that one.

**Comment by alexandrudima** [comment]
Does this work in the editor at https://microsoft.github.io/monaco-editor/ ? (uses the same code for input handling).

**Comment by myfrom** [comment]
@alexandrudima Yup, it works in those input fields. Currently at Chrome 68.0.3440.75

**Comment by alexandrudima** [comment]
VSCode stable is currently on Chromium 58.0.3029.110 and VSCode insiders is on Chromium 61.0.3163.100

**Comment by myfrom** [comment]
I mean that I opened it in Chrome 68.

Do you think it is a problem of older Chromium?

**Comment by alexandrudima** [comment]
Yes, very likely.

**Comment by neclepsio** [comment]
I have a M535 mouse. It works in Chrome, but "Logitech Options", the software that manages the configuration of the mouse, tweaks the messages the mouse sends according to the application. For example, Excel does not handle the "button scroll left", so the driver sends a direct message to scroll. Maybe that is the culprit. Horizontal scrolling worked for me in VSCode 1.21, I cannot tell if it's the Options update or VSCode update that broke it. Sometimes, not always, opening Options solves for VSCode (Chrome always works).

**Comment by alexandrudima** [comment]
> tweaks the messages the mouse sends according to the application
... that is ... not good ...

**Comment by SanwariyaAmit** [comment]
I'm using the Logitech M720 Triathlon, facing the same issue.
There's a jerk (brief movement) on trying to scroll but then it stops responding completely.

More Details:
- Horizontal Scrolling on mouse doesn't works in VS Code stable, insider and exploration builds.
- Also doesn't works in [Electron Fiddle ](https://github.com/electron/fiddle/releases/download/0.1/electron-fiddle-0.1.0-setup-win32-x64.exe) suggested by @bpasero 
- Works fine on the Monaco editor page: https://microsoft.github.io/monaco-editor/
- `window.smoothScrollingWorkaround: true` doesn't helps at all.
- Minimizing and restoring window fixes it, until resized.

**Comment by apolosky** [comment]
Using the MX Master mouse and I'm having this issue as well.

**Comment by alexandrudima** [comment]
> It works in Chrome, but "Logitech Options", the software that manages the configuration of the mouse, tweaks the messages the mouse sends according to the application

You need to reach out to the manufacturers of the mice if the messages are tweaked according to the application, there is nothing VS Code can do if we don't receive mouse wheel events.

**Comment by apaatsio** [comment]
For what it's worth, I have Logitech MX Master and horizontal scrolling is *working fine on Linux*.

**Comment by Adunaphel** [comment]
Using MX Master on both Windows 10 (1803 and 1809) and Mac OS Mojave, horizontal scroll works fine on Mac OS, not working on Windows.
* Working fine on Monaco https://microsoft.github.io/monaco-editor/
* Not working in Electron Fiddle
* Both machines are using stable branches of VS Code
* Logitech Options on newest version
* No per app settings configured
* OSes kept up to date

**Comment by BeatMercy** [comment]
now my mouse works fine with the newest  Logitech Options........on Windows 1809😂
Actually, my problem solved since the logitech options update months ago

**Comment by BeatMercy** [comment]
If problem still haunting around, see also -> https://github.com/Microsoft/vscode/issues/56964

**Comment by SanwariyaAmit** [comment]
Issue still occurs on 1809.

**Comment by myfrom** [comment]
@beatmercy Could you post the version of Logitech Options that you're on?

**Comment by BeatMercy** [comment]
- Software: Logitech Options version is 6.92.275
- OS: Window 1809
- Laptop: Asus FX50j
hope it help~~

**Comment by myfrom** [comment]
I also still can't get it to work 🙁 I'm on Windows 1803 so maybe that's the problem but I doubt it. I'll just subscribe to the other issue and hope for a fix to come soon :wink:

**Comment by slater101** [comment]
I know this is closed, but same issue here lol. Doesnt happen when the logitech options is not installed. So yea... Something there. Works fine without the software. 

With options installed, reacts strange, sometimes fine for seconds, then all of a sudden no response. 

Works in every other program I have ever used, including all adobe products, browsers, ( opera, IE 11, Moz, Chrome ) Works fine in Atom editor etc.. list goes on forever, never seen this happen on any other program.
